megatron/training/training.py

```diff
@@ -1171,15 +1174,19 @@ def get_megatron_optimizer_config(args: Any) -> OptimizerConfig:
 
     # Construct the appropriate config_overrides object.
     # TODO: add more logic here as needed down the road.
+    config_overrides: Optional[Dict[ParamKey, ParamGroupOverride]] = {}
     if args.decoupled_lr is not None:
+        decoupled_lr_config: ParamGroupOverride = {"max_lr": args.decoupled_lr}
         decoupled_param_key = ParamKey(attr="is_embedding_or_output_parameter")
-        decoupled_optimizer_config = copy.deepcopy(config)
-        decoupled_optimizer_config.lr = args.decoupled_lr
         if args.decoupled_min_lr is not None:
-            decoupled_optimizer_config.min_lr = args.decoupled_min_lr
-        config_overrides = {decoupled_param_key: decoupled_optimizer_config}
-    else:
-        config_overrides = None
+            decoupled_lr_config["min_lr"] = args.decoupled_min_lr
+        config_overrides[decoupled_param_key] = decoupled_lr_config
+    
+    # Next construct the standard param group overrides for no weight decay on bias parameters 
+    #  as well as any length 1 parameters.
+    param_length_1_match = ParamPredicate(name="param_len_1", fn=lambda param: len(param.shape) == 1)
+    param_wd_mult_key = ParamKey(name="*.bias", predicate=param_length_1_match)
+    config_overrides[param_wd_mult_key] = ParamGroupOverride(wd_mult=0.0)
```

Comment:
@ko3n1g this is the code block where I am implementing the previous weight decay rule. Hopefully this is flexible enough for the model you are blocked on. LMK if not.

Reply:
I wonder if this should go into MCore somehow to make sure all consumers of MCore don't see changed default behavior?